core/settings.py

```
@@ -35,7 +35,16 @@
     DEBUG = os.environ.get('DEBUG', 'True').lower() == 'true'
 
-ALLOWED_HOSTS = ['.vercel.app', '*']
+ALLOWED_HOSTS = ['.vercel.app']
+
+if DEBUG or not IS_PRODUCTION:
+    ALLOWED_HOSTS.extend(['localhost', '127.0.0.1'])
+
+# Add extra domains from ALLOWED_HOSTS_EXTRA environment variable
+extra_hosts = [host.strip() for host in os.environ.get('ALLOWED_HOSTS_EXTRA', '').split(',') if host.strip()]
+if '*' in extra_hosts:
+    raise ImproperlyConfigured("ALLOWED_HOSTS_EXTRA must not contain '*'")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)